youtube_dlc/downloader/youtube_live_chat.py

```diff
@@ -43,23 +47,35 @@ def parse_yt_initial_data(data):
                 except AttributeError:
                     continue
 
-        def download_and_parse_fragment(url, frag_index):
+        def parse_ytcfg(data):
+            data = data.decode('utf-8', 'replace')
+            try:
+                raw_json = re.search(r'ytcfg\.set\s*\(\s*({.+?})\s*\)\s*;', data).group(1)
+                return json.loads(raw_json)
+            except IndexError:
+                return None
+
```

Comment:
Those are protected methods if I understand the underscore, but I addressed this in https://github.com/pukkandan/yt-dlp/pull/86/commits/255277656d2b927bf3ee30e2c4fd7dabab3d6b36

Reply:
> Those are protected methods if I understand the underscore

Technically, yes. Ideally, the live-chat extraction should happen inside the youtube extractor itself (like comments). But since you already implemented it as a downloader, it'd be a headache to do that now